source/steamshim/src/steamshim_private.h

```diff
@@ -41,6 +43,7 @@ extern PipeType GPipeWrite;
 class pipebuff_t
 {
   public:
+  pipebuff_t();
   char buffer[PIPEMESSAGE_MAX];
   unsigned int cursize = 0;
```

Comment:
Should this be public?
Same with `hasmsg` and `br` (whatever that is).

You seem to be using variables for other stuff than their name suggests (f.e. `cursize` for a read pointer)